src/client.rs

```diff
@@ -1286,6 +1286,54 @@ impl<T: Read + Write> Session<T> {
             .and_then(|lines| parse_ids(&lines, &mut self.unsolicited_responses_tx))
     }
 
+    /// The SORT command is a variant of SEARCH with sorting semantics for
+    /// the results. There are two arguments before the searching
+    /// criteria argument: a parenthesized list of sort criteria, and the
+    /// searching charset.
+    ///
+    /// Below is a selection of common sort keys. The full list can be found in the
+    /// specification of the [`SORT command`](https://tools.ietf.org/html/rfc5256#section-3):
+    ///  - `ARRIVAL`: Internal date and time of the message
+    ///  - `CC`: IMAP addr-mailbox of the first "cc" address
+    ///  - `DATE`: Sent date and time
+    ///  - `FROM`: IMAP addr-mailbox of the first "From" address
+    ///  - `REVERSE <sort-key>`: Followed by another sort criterion, has the effect of that criterion but in reverse (descending) order
+    ///  - `SIZE`: Size of the message in octets
+    ///  - `SUBJECT`: Base subject text
+    ///  - `TO`: IMAP addr-mailbox of the first "To" address
+    pub fn sort<'s, S: AsRef<str>>(
+        &mut self,
+        criteria: &'s [extensions::sort::SortCriterion<'s>],
+        charset: extensions::sort::SortCharset<'s>,
+        query: S,
+    ) -> Result<Vec<Seq>> {
+        self.run_command_and_read_response(&format!(
+            "SORT {} {} {}",
+            extensions::sort::SortCriteria(criteria),
+            charset,
+            query.as_ref()
+        ))
+        .and_then(|lines| parse_ordered_ids(&lines, &mut self.unsolicited_responses_tx))
+    }
+
+    /// Equivalent to [`Session::sort`], except that the returned identifiers
+    /// are [`Uid`] instead of [`Seq`]. See also the [`UID`
+    /// command](https://tools.ietf.org/html/rfc3501#section-6.4.8).
```

Comment:
In Rust, the first paragraph is considered a summary, and often rendered on its own. Let's take advantage of that and make this:
```suggestion
    /// Equivalent to [`Session::sort`], except that it returns [`Uid`]s.
    ///
    /// See also [`Session::uid_search`].
```

Reply:
> In Rust, the first paragraph is considered a summary, and often rendered on its own.

I did not know about this, thank you :+1: 

:heavy_check_mark: Done